src/network/network_gui.cpp

```diff
@@ -1894,7 +1894,7 @@ struct NetworkClientListWindow : Window {
 
 		if (_network_server) this->buttons[line_count].emplace_back(new CompanyButton(SPR_ADMIN, STR_NETWORK_CLIENT_LIST_ADMIN_COMPANY_TOOLTIP, COLOUR_RED, company_id, &NetworkClientListWindow::OnClickCompanyAdmin, company_id == COMPANY_SPECTATOR));
 		this->buttons[line_count].emplace_back(chat_button);
-		if (own_ci->client_playas != company_id) this->buttons[line_count].emplace_back(new CompanyButton(SPR_JOIN, STR_NETWORK_CLIENT_LIST_JOIN_TOOLTIP, COLOUR_ORANGE, company_id, &NetworkClientListWindow::OnClickCompanyJoin));
+		if (own_ci->client_playas != company_id && (company_id == COMPANY_SPECTATOR || !Company::Get(company_id)->is_ai)) this->buttons[line_count].emplace_back(new CompanyButton(SPR_JOIN, STR_NETWORK_CLIENT_LIST_JOIN_TOOLTIP, COLOUR_ORANGE, company_id, &NetworkClientListWindow::OnClickCompanyJoin));
```

Comment:
As you mention yourself, admins can join it just fine. So I guess we should add that exception in here too?

Reply:
`join` console command doesn't allow joining AI company, even for admins.

Reply:
That is just hateful. What did the console ever do to the admins? Tssk.

Reply:
After some talk on IRC, lets disable it instead of removing. That is simply moving this change, as `CompanyButton` has an additional parameter to indicate disabled state. See line 1895 for an example.